pandas/tests/tools/test_to_datetime.py

```diff
@@ -463,14 +468,14 @@ def test_to_datetime_parse_timezone_keeps_name(self):
 class TestToDatetime:
     def test_to_datetime_mixed_datetime_and_string(self):
         # GH#47018 adapted old doctest with new behavior
-        d1 = datetime(2020, 1, 1, 17, tzinfo=timezone(-timedelta(hours=1)))
         d2 = datetime(2020, 1, 1, 18, tzinfo=timezone(-timedelta(hours=1)))
-        res = to_datetime(["2020-01-01 17:00 -0100", d2])
-        expected = to_datetime([d1, d2]).tz_convert(pytz.FixedOffset(-60))
-        tm.assert_index_equal(res, expected)
+        with pytest.raises(
+            ValueError,
+            match=r"time data '.*' does not match format '%Y-%m-%d %H:%M %z' \(match\)",
+        ):
+            to_datetime(["2020-01-01 17:00 -0100", d2])
```

Comment:
to me, this seems like a loss in functionality.  We should be able to convert a mix of strings and `datetime` objects.